pkgs/servers/alice-lg/default.nix

```diff
@@ -0,0 +1,85 @@
+{ lib
+, fetchFromGitHub
+, buildGoModule
+, fetchYarnDeps
+, stdenv
+, yarn
+, nodejs
+, git
+, fixup_yarn_lock
+}:
+
+buildGoModule rec {
+  pname = "alice-lg";
+  version = "6.0.0";
+
+  src = fetchFromGitHub {
+    owner = "alice-lg";
+    repo = "alice-lg";
+    rev = version;
+    hash = "sha256-BdhbHAFqyQc8UbVm6eakbVmLS5QgXhr06oxoc6vYtsM=";
+  };
+
+  vendorSha256 = "sha256-SNF46uUTRCaa9qeGCfkHBjyo4BWOlpRaTDq+Uha08y8=";
+
+  ui = stdenv.mkDerivation {
+    pname = "alice-lg-ui";
+    src = "${src}/ui";
+    inherit version;
+
+    yarnOfflineCache = fetchYarnDeps {
+      yarnLock = src + "/ui/yarn.lock";
+      hash = "sha256-NeK9IM8E2IH09SVH9lMlV3taCmqwlroo4xzmv4Q01jI=";
+    };
+
+    nativeBuildInputs = [ nodejs yarn git ];
+    configurePhase = ''
+      runHook preConfigure
+
+      # Yarn and bundler wants a real home directory to write cache, config, etc to
+      export HOME=$NIX_BUILD_TOP/fake_home
+
+      # Make yarn install packages from our offline cache, not the registry
+      yarn config --offline set yarn-offline-mirror $yarnOfflineCache
+
+      # Fixup "resolved"-entries in yarn.lock to match our offline cache
+      ${fixup_yarn_lock}/bin/fixup_yarn_lock yarn.lock
+
+      yarn install --offline --frozen-lockfile --ignore-scripts --no-progress --non-interactive
+      patchShebangs node_modules/
+      runHook postConfigure
+    '';
+
+    buildPhase = ''
+      runHook preBuild
+
+      ./node_modules/.bin/react-scripts build
+
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      ls -la
+      mv build $out
+
+      runHook postInstall
+    '';
+  };
+
+  preBuild = ''
+    cp -R ${ui}/ ui/build/
```

Comment:
does this have to be a `cp` rather than a `lndir`/`ln`?

Reply:
lndir doesn't exist on my system and `ln -s -d ${ui} ui/build` for a symbolic link of the directory causes the go build to throw an error: `ui/assets.go:12:12: pattern build/*: cannot embed file build/asset-manifest.json: in non-directory build`